### ISSUE TITLE
Captcha

### ISSUE THREAD

**Issue by ghost** [shubhams167/chegg-pybot#1]
Any idea on how to prevent getting the captcha in the first place?

**Comment by shubhams167** [comment]
There are a few tips to prevent getting detected by bot management tools but unfortunately, none of them are 100% effective. Some of the tips are: 
- Use maximized browser window
- Don't use the _headless_ option
- Modifying the _navigator.webdriver_ flag

I am working on this bot to behave more and more like a normal human would behave on the website by adding delays before executing commands. 
In short, if your bot behaves like a normal human instead of just clicking on links instantly then I guess it would not get flagged.

**Comment by ghost** [comment]
Can you expand on Modifying the navigator.webdriver flag.

I tried a bunch and I mean a bunch of stuff like setting user-profile, useragent, maximum size browser, random times, etc but I still end up getting Catpcha. I even changed the hex of the chromedriver.

**Comment by shubhams167** [comment]
You could change the JavaScript variable `navigator.webdriver` which is set to `true` when a browser is controlled by automation tools. But I don't think this will do the trick. Captchas are really hard to prevent otherwise what's the purpose of captcha?

You can try to automate the solving of captchas by automating a click on Google's reCaptcha. If it accepts and goes away with just a click, then good otherwise you might need to try a few services like [deathbycaptcha](https://www.deathbycaptcha.com/).
